packer/builder/azure/arm/builder.go

```diff
@@ -57,28 +60,67 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	servicePrincipalTokenVault, err := b.createServicePrincipalTokenVault()
+	if err != nil {
+		return nil, err
+	}
+
 	ui.Message("Creating Azure Resource Manager (ARM) client ...")
-	azureClient, err := NewAzureClient(b.config.SubscriptionID, b.config.ResourceGroupName, b.config.StorageAccount, servicePrincipalToken)
+	azureClient, err := NewAzureClient(b.config.SubscriptionID, b.config.ResourceGroupName, b.config.StorageAccount, servicePrincipalToken, servicePrincipalTokenVault)
 	if err != nil {
 		return nil, err
 	}
 
-	steps := []multistep.Step{
-		NewStepCreateResourceGroup(azureClient, ui),
-		NewStepValidateTemplate(azureClient, ui),
-		NewStepDeployTemplate(azureClient, ui),
-		NewStepGetIPAddress(azureClient, ui),
-		&communicator.StepConnectSSH{
-			Config:    &b.config.Comm,
-			Host:      lin.SSHHost,
-			SSHConfig: lin.SSHConfig(b.config.UserName),
-		},
-		&common.StepProvision{},
-		NewStepGetOSDisk(azureClient, ui),
-		NewStepPowerOffCompute(azureClient, ui),
-		NewStepCaptureImage(azureClient, ui),
-		NewStepDeleteResourceGroup(azureClient, ui),
-		NewStepDeleteOSDisk(azureClient, ui),
+	var steps []multistep.Step
+
+	if b.config.OSType == constants.GOOS_Linux {
```

Comment:
GOOS_Linux is a constant to compare to the GOOS constant. Please don't reuse it for other purposes.